app/controllers/addresses_controller.rb

```diff
@@ -0,0 +1,45 @@
+class AddressesController < ApplicationController
+  def index 
+    @address = Address.new 
+  end
+
+  def edit
+    @address = Address.find_by(id: params[:id])
+    redirect_to root_path, notice: "Address Not Found!" if @address.blank?
```

Comment:
You need to handle the record not found case for other actions as well. eg. update, destroy..
And try to change the id in the browser url on the edit page. I think anyone can edit the other user's address and even without login.